packages/ckeditor5-cloud-services/src/cloudservices.ts

```diff
@@ -0,0 +1,286 @@
+ * @module cloud-services/cloudservices
+ */
+
+import { ContextPlugin, type ContextPluginDependencies } from 'ckeditor5/src/core';
+import { CKEditorError } from 'ckeditor5/src/utils';
+import CloudServicesCore from './cloudservicescore';
+import type { InitializedToken } from './token/token';
+
+/**
+ * Plugin introducing the integration between CKEditor 5 and CKEditor Cloud Services .
+ *
+ * It initializes the token provider based on
+ * the {@link module:cloud-services/cloudservices~CloudServicesConfig `config.cloudService`}.
+ */
+export default class CloudServices extends ContextPlugin implements CloudServicesConfig {
+	/**
+	 * The authentication token URL for CKEditor Cloud Services or a callback to the token value promise. See the
+	 * {@link module:cloud-services/cloudservices~CloudServicesConfig#tokenUrl} for more details.
+	 */
+	public readonly tokenUrl!: TokenUrl;
+
+	/**
+	 * The URL to which the files should be uploaded.
+	 */
+	public readonly uploadUrl?: string;
+
+	/**
+	 * The URL for web socket communication, used by the `RealTimeCollaborativeEditing` plugin. Every customer (organization in the CKEditor
+	 * Ecosystem dashboard) has their own, unique URLs to communicate with CKEditor Cloud Services. The URL can be found in the
+	 * CKEditor Ecosystem customer dashboard.
+	 *
+	 * Note: Unlike most plugins, `RealTimeCollaborativeEditing` is not included in any CKEditor 5 build and needs to be installed manually.
+	 * Check [Collaboration overview](https://ckeditor.com/docs/ckeditor5/latest/features/collaboration/overview.html) for more details.
+	 */
+	public readonly webSocketUrl?: string;
+
+	/**
+	 * An optional parameter used for integration with CKEditor Cloud Services when uploading the editor build to cloud services.
+	 *
+	 * Whenever the editor build or the configuration changes, this parameter should be set to a new, unique value to differentiate
+	 * the new bundle (build + configuration) from the old ones.
+	 */
+	public readonly bundleVersion?: string;
+
+	/**
+	 * Other plugins use this token for the authorization process. It handles token requesting and refreshing.
+	 * Its value is `null` when {@link module:cloud-services/cloudservices~CloudServicesConfig#tokenUrl} is not provided.
+	 *
+	 * @readonly
+	 */
+	public token: InitializedToken | null = null;
+
+	/**
+	 * A map of token object instances keyed by the token URLs.
+	 */
+	private readonly _tokens = new Map<TokenUrl, InitializedToken>();
+
+	/**
+	 * @inheritdoc
+	 */
+	public static get pluginName(): 'CloudServices' {
+		return 'CloudServices';
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public static get requires(): ContextPluginDependencies {
+		return [ CloudServicesCore ];
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public async init(): Promise<void> {
+		const config = this.context.config;
+
+		const options: any = config.get( 'cloudServices' ) || {};
+
+		for ( const optionName in options ) {
+			( this as any )[ optionName ] = options[ optionName ];
+		}
```

Comment:
```suggestion
		const options = config.get( 'cloudServices' ) || {};
		
		for ( const [ key, value ] of Object.entries( options ) as Array<[key: keyof CloudServicesConfig, value: any]> ) {
			( this as CloudServicesConfig )[ key ] = value;
		}
```

If you'd like to get rid of casting to `any` you can try the code above, but I think it would make it less readable anyway.